browser/ui/brave_browser.cc

```diff
@@ -41,6 +46,23 @@ namespace {
 
 bool g_suppress_dialog_for_testing = false;
 
+bool IsNTP(content::WebContents* web_contents) {
+  // Use the committed entry so the bookmarks bar disappears at the same time
+  // the page does.
+  content::NavigationEntry* entry =
```

Comment:
We should not be copying the logic from `IsShowingNTP`, copying code is worse than patching. Also this is missing a check for sad tab that the upstream method has an is named differently which makes it even more confusing

Reply:
cc: @cdesouza-chromium

Reply:
I guess once I understand https://github.com/brave/brave-core/pull/28944/commits/f49fb39dad97133d5a1a2841119a9b67f73040bd#r2132743252 we can figure out what to do with this

Reply:
Fixed in 459df983e3f